src/storeController/index.ts

```diff
@@ -104,6 +107,15 @@ export default async function (
       }
     }
   }
+
+  async function upload (nodeVersion: string, pkg: {peripheralLocation: string, id: string}) {
+    const nodeMajor = nodeVersion.substring(0, nodeVersion.indexOf('.'))
+    const cachePath = path.join(store, 'side-effects-cache', nodeMajor, pkg.id)
+    if (await fs.exists(cachePath) || await fs.exists(`${cachePath}+stage`)) {
+      return
```

Comment:
Why not rewrite in this case?

Reply:
because if the cache is already present, or is in the process of being copied, we shouldn't copy again. It saves time.

Reply:
what if the already present cache is corrupt? how to update it?

Reply:
I don't know what would be the best way to fix it... suggestions? Re-copying the side effects cache every time sounds time consuming to me, and a bit dangerous: say that there are two concurrent installations going on, one installation would override the cache of the other one, actually _causing_ corruptions.